packages/story-editor/src/components/canvas/rightClickMenu.js

```diff
@@ -89,19 +157,30 @@ const RightClickMenu = () => {
             isRTL={isRTL}
           >
             {rightClickMenuItems.map(
-              ({ label, shortcut, separator, ...buttonProps }) => (
+              ({
+                label,
+                shortcut,
+                separator,
+                subMenuItems,
+                ...buttonProps
+              }) => (
                 <Fragment key={label}>
                   {separator === 'top' && (
                     <ContextMenuComponents.MenuSeparator />
                   )}
-                  <ContextMenuComponents.MenuButton {...buttonProps}>
-                    {label}
-                    {shortcut && (
-                      <ContextMenuComponents.MenuShortcut>
-                        {shortcut.display}
-                      </ContextMenuComponents.MenuShortcut>
+                  <ButtonWrapper>
+                    {menuButtonRenderer({ label, shortcut, ...buttonProps })}
+                    {Boolean(subMenuItems?.length) && (
+                      <SubMenuWrapper
+                        isRTL={isRTL}
+                        aria-expanded
+                        isOpen
+                        isSubMenu
+                      >
+                        {subMenuItems.map((item) => menuButtonRenderer(item))}
+                      </SubMenuWrapper>
                     )}
-                  </ContextMenuComponents.MenuButton>
+                  </ButtonWrapper>
```

Comment:
💭 I think we don't need to do this.

The goal of the refactor was to be able to create context menus with more flexibility. Included in this was the idea that we could _nest_ context menus together.

```js

<RightClickMenuContainer position={menuPosition} ref={ref}>
  <DirectionAware>
    <ContextMenu
      data-testid="right-click-context-menu"
      isOpen={isMenuOpen}
      onDismiss={onCloseMenu}
      aria-label={__('Context Menu for the selected element', 'web-stories')}
      maskRef={maskRef}
      onMouseDown={(evt) => evt.stopPropagation()}
      isRTL={isRTL}
    >
      {rightClickMenuItems.map(
        ({ label, shortcut, separator, subMenuItems, ...buttonProps }) => (
          <Fragment key={label}>
            {separator === 'top' && <ContextMenuComponents.MenuSeparator />}
            <ContextMenuComponents.MenuButton {...buttonProps}>
              {label}
              {shortcut && (
                <ContextMenuComponents.MenuShortcut>
                  {shortcut.display}
                </ContextMenuComponents.MenuShortcut>
              )}
            </ContextMenuComponents.MenuButton>
            {Boolean(subMenuItems?.length) && (
               {/* The position of the new menu is defined here */}
              <RightClickMenuContainer position={{ y: 0, x: 430 }}>
                <ContextMenu
                  data-testid="right-click-context-menu"
                  isOpen={isMenuOpen}
                  // No `onDismiss` so clicking the layer buttons won't close the menu
                  // if that is the desired behavior
                  aria-label={__(
                    'Select a layer',
                    'web-stories'
                  )}
                  onMouseDown={(evt) => evt.stopPropagation()}
                  isRTL={isRTL}
                >
                  {subMenuItems.map(
                    ({
                      label: subLabel,
                      shortcut: subShortcut,
                      icon,
                      supportsIcon,
                      SuffixIcon,
                      ...nestedButtonProps
                    }) => (
                      <ContextMenuComponents.MenuButton
                        key={subLabel}
                        {...nestedButtonProps}
                      >
                        <ContextMenuComponents.MenuLabel>
                          {icon}
                          {subLabel}
                        </ContextMenuComponents.MenuLabel>
                        {shortcut && (
                          <ContextMenuComponents.MenuShortcut>
                            {subShortcut.display}
                          </ContextMenuComponents.MenuShortcut>
                        )}
                        {SuffixIcon && (
                          <SuffixIconWrapper>
                            <SuffixIcon />
                          </SuffixIconWrapper>
                        )}
                      </ContextMenuComponents.MenuButton>
                    )
                  )}
                </ContextMenu>
              </RightClickMenuContainer>
            )}
            {separator === 'bottom' && <ContextMenuComponents.MenuSeparator />}
          </Fragment>
        )
      )}
    </ContextMenu>
  </DirectionAware>
</RightClickMenuContainer>;
```

The above code should render the menu without needing the extra styles:
![image](https://user-images.githubusercontent.com/22185279/150138073-53e0f9db-d70d-45e5-aaad-88401971f5fb.png)

Some massaging will be needed (like adding an icon to the button that expands the sub menu, and accessibility work).

Was there a reason that you didn't want to do it this way that I'm missing?

Reply:
Hm, I tried originally with using a second `ContextMenu` but it didn't work as expected -- wasn't aware of the `RightClickMenuContainer` though! Probably that's why.

I'd still create a separate menuItemRenderer though that's shared between the menus since they're using exactly the same style. Any reason why you think it's better not to?

Reply:
Yeah that's the magic touch! Just setting the position using the containing div should work.

`menuItemRenderer`: Sure that's fine! Maybe we could even make this its own component?